src/float.cr

```diff
@@ -148,8 +148,17 @@ struct Float32
     Printer.print(self, io)
   end
 
+  private def hash_normalize
+    t = to_i32
+    t == self ? t : unsafe_as(Int32)
```

Comment:
@funny-falcon Float hashes in `pyhash` manner have identical values to integer values if no fraction part exists. Fell free to include this, of course.